### PR TITLE
Course Problem List Implementation

### PR DESCRIPTION
**Pull request by abhishek2021005**
=> Among ProblemList, ProblemList1, ProblemList2,ProblemList3,ProblemList4 any one problemlist will be finally rendered and other should be commented out in practice-problem pages according to the choice.

=> And also apart from problemlist component,per-section-quiz page, and practice-problems page only a minor change is done in course-home pages (in which a Practice Button is added line number :- 171).

=> Remaining all files are unchanged only  code is refactored b/c of default prettier mode in my vscode.

### CODE REVIEW

File: packages/alea-frontend/pages/course-home/[courseId].tsx [hunk 1]
```diff
@@ -190,6 +168,10 @@ const CourseHomePage: NextPage = () => {
             {t.studyBuddy}&nbsp;
             <Diversity3Icon fontSize="large" />
           </CourseComponentLink>
+          <CourseComponentLink href={`/practice-problems/${courseId}`}>
+            {<p>Practice</p>}&nbsp;
```
Comment (XI4507) [inline packages/alea-frontend/pages/course-home/[courseId].tsx:172]: Do localization of "Practice"

File: packages/alea-frontend/pages/course-problems/[courseId].tsx [hunk 2]
```diff
@@ -33,15 +28,11 @@ const CourseProblemsPage: NextPage = () => {
     router.replace('/');
     return <>Course Not Found!</>;
   }
-  const url = XhtmlContentUrl(
-    courseInfo.notesArchive,
-    courseInfo.notesFilepath
-  );
+  const url = XhtmlContentUrl(courseInfo.notesArchive, courseInfo.notesFilepath);
+  console.log('uR', url);
```
Comment (XI4507) [inline packages/alea-frontend/pages/course-problems/[courseId].tsx:32]: remove console.log

File: packages/alea-frontend/pages/course-home/[courseId].tsx (hunk 1 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/course-home/[courseId].tsx]: Localization of all hardcoded value done.

File: packages/alea-frontend/pages/course-problems/[courseId].tsx (hunk 2 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/course-problems/[courseId].tsx]: Removed.

File: packages/alea-frontend/pages/per-section-quiz.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                backgroundColor: PRIMARY_COL,
+                color: 'white',
+              },
+            }}
+            onClick={handleButtonClick}
+          >
+            {t.courseProblemPage}&nbsp;
+          </Button>
+        </Tooltip>
+        {/* <Box display="flex" flexDirection="column" alignItems="center">
```
Comment (XI4507) [inline packages/alea-frontend/pages/per-section-quiz.tsx:53]: remove this comment

File: packages/alea-frontend/pages/per-section-quiz.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  const handleButtonClick = () => {
+    router.push(`/practice-problems/${courseId}`);
+  };
+
+  if (!archive || !filepath) {
+    return <div>No data found</div>;
+  }
+
+  return (
+    <MainLayout title="All Comments | VoLL-KI">
```
Comment (XI4507) [inline packages/alea-frontend/pages/per-section-quiz.tsx:30]: Why title = All comments?

File: packages/alea-frontend/pages/per-section-quiz.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+import { Typography, Button, Box, Tooltip } from '@mui/material';
+import { PerSectionQuiz } from 'packages/stex-react-renderer/src/lib/PerSectionQuiz';
+import MainLayout from '../layouts/MainLayout';
+import { PRIMARY_COL } from '@stex-react/utils';
+import { getLocaleObject } from 'packages/stex-react-renderer/src/lib/lang/utils';
+// import { getLocaleObject } from '../lang/utils';
```
Comment (XI4507) [inline packages/alea-frontend/pages/per-section-quiz.tsx:8]: remove the commented line

File: packages/alea-frontend/pages/practice-problems/[courseId].tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+  if (!data) return <CircularProgress />;
+
+  return (
+    <MainLayout title={`${(courseId || '').toUpperCase()} ${t.notes} | VoLL-KI`}>
+      <ProblemList data={data} courseId={courseId} />
+      <ProblemList3 data={data} courseId={courseId} />
+      <ProblemList4 data={data} courseId={courseId} />
+      <ProblemList1 data={data} courseId={courseId} />
+      <ProblemList2 data={data} courseId={courseId} />
```
Comment (XI4507) [inline packages/alea-frontend/pages/practice-problems/[courseId].tsx:69]: remove unwanted problemlist component

File: packages/alea-frontend/pages/per-section-quiz.tsx (hunk 5 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/per-section-quiz.tsx]: Done.

File: packages/alea-frontend/pages/per-section-quiz.tsx (hunk 4 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/per-section-quiz.tsx]: Changed title according to page shown

File: packages/alea-frontend/pages/per-section-quiz.tsx (hunk 3 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/per-section-quiz.tsx]: done

File: packages/alea-frontend/pages/practice-problems/[courseId].tsx (hunk 6 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/practice-problems/[courseId].tsx]: Removed.

File: packages/alea-frontend/pages/practice-problems/[courseId].tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import { useContext, useEffect, useState } from 'react';
+import { getLocaleObject } from '../../lang/utils';
+import MainLayout from '../../layouts/MainLayout';
+import ProblemList from 'packages/alea-frontend/components/ProblemList';
+import ProblemList1 from 'packages/alea-frontend/components/ProblemList1';
+import ProblemList2 from 'packages/alea-frontend/components/ProblemList2';
+import ProblemList3 from 'packages/alea-frontend/components/ProblemList3';
+import ProblemList4 from 'packages/alea-frontend/components/ProblemList4';
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/alea-frontend/pages/practice-problems/[courseId].tsx]: Done.

File: packages/alea-frontend/components/ProblemList.tsx [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+        <List>
+          {titlesAndMetadata.map((item, index) => {
+            const problemCount = problemCounts[item.id] || 0;
+            const isBold = item.level === 2;
+            const fontWeight = isBold ? 'bold' : 'normal';
+            const backgroundColor = item.level % 2 === 0 ? '#f0f4f8' : 'blue';
+            const borderRadius = item.level % 2 === 0 ? '8px' : '0px';
+            const fontFamily =
+              item.level === 2 ? 'Times New Roman, Times, serif' : 'Roboto, sans-serif';
+
```
Comment (XI4507) [inline packages/alea-frontend/components/ProblemList.tsx:163]: Can we write a separate  function for this

Comment (abhishek2021005) [reply packages/alea-frontend/components/ProblemList.tsx:163]: Yeah we can write a separate function but i think in this particular case having related logic inline within map is better and also overhead of calling a function will not be there for every iteration of map on each item.

**Pull request merged**